core/src/main/java/com/datastrato/gravitino/storage/relational/session/SqlSessionFactoryHelper.java

```diff
@@ -0,0 +1,110 @@
+import com.datastrato.gravitino.Configs;
+import com.datastrato.gravitino.storage.relational.mapper.MetalakeMetaMapper;
+import com.google.common.base.Preconditions;
+import java.sql.SQLException;
+import java.time.Duration;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.commons.pool2.impl.BaseObjectPoolConfig;
+import org.apache.ibatis.mapping.Environment;
+import org.apache.ibatis.session.Configuration;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.apache.ibatis.session.SqlSessionFactoryBuilder;
+import org.apache.ibatis.transaction.TransactionFactory;
+import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory;
+
+/**
+ * SqlSessionFactoryHelper maintains the MyBatis's {@link SqlSessionFactory} object, which is used
+ * to create the {@link org.apache.ibatis.session.SqlSession} object. It is a singleton class and
+ * should be initialized only once.
+ */
+public class SqlSessionFactoryHelper {
+  private static volatile SqlSessionFactory sqlSessionFactory;
+  private static final SqlSessionFactoryHelper INSTANCE = new SqlSessionFactoryHelper();
+
+  public static SqlSessionFactoryHelper getInstance() {
+    return INSTANCE;
+  }
+
+  private SqlSessionFactoryHelper() {}
+
+  /**
+   * Initialize the SqlSessionFactory object.
+   *
+   * @param config Config object to get the jdbc connection details from the config.
+   */
+  @SuppressWarnings("deprecation")
+  public void init(Config config) {
+    // Initialize the data source
+    BasicDataSource dataSource = new BasicDataSource();
+    dataSource.setUrl(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_URL));
+    dataSource.setDriverClassName(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_DRIVER));
+    dataSource.setUsername(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_USER));
+    dataSource.setPassword(config.get(Configs.ENTITY_RELATIONAL_JDBC_BACKEND_PASSWORD));
+    // Close the auto commit, so that we can control the transaction manual commit
+    dataSource.setDefaultAutoCommit(false);
+    dataSource.setMaxWaitMillis(1000L);
+    dataSource.setMaxTotal(20);
+    dataSource.setMaxIdle(5);
+    dataSource.setMinIdle(0);
+    dataSource.setLogAbandoned(true);
+    dataSource.setRemoveAbandonedOnBorrow(true);
+    dataSource.setRemoveAbandonedTimeout(60);
+    dataSource.setTimeBetweenEvictionRunsMillis(Duration.ofMillis(10 * 60 * 1000L).toMillis());
+    dataSource.setTestOnBorrow(BaseObjectPoolConfig.DEFAULT_TEST_ON_BORROW);
+    dataSource.setTestWhileIdle(BaseObjectPoolConfig.DEFAULT_TEST_WHILE_IDLE);
+    dataSource.setMinEvictableIdleTimeMillis(1000);
+    dataSource.setNumTestsPerEvictionRun(BaseObjectPoolConfig.DEFAULT_NUM_TESTS_PER_EVICTION_RUN);
+    dataSource.setTestOnReturn(BaseObjectPoolConfig.DEFAULT_TEST_ON_RETURN);
+    dataSource.setSoftMinEvictableIdleTimeMillis(
+        BaseObjectPoolConfig.DEFAULT_SOFT_MIN_EVICTABLE_IDLE_TIME.toMillis());
+    dataSource.setLifo(BaseObjectPoolConfig.DEFAULT_LIFO);
```

Comment:
Do you want to make several of these parameters configurable from outside?

Reply:
I think we can not provide parameter configuration at present, otherwise there may be too many parameters to manage. Maybe it would be a good idea to provide some parameters on demand afterwards